tests/test_main.py

```diff
@@ -142,20 +143,52 @@ def test_redo_deps_if_reverted(gitrepo):
     # make sure a and b were each run exactly once
     entries = env['LOGGER'].load()
     a_ends = [e for e in entries if e['activity'] == 'a' and
-                                    e['category'] == 'activity-end']
+              e['category'] == 'activity-end']
     assert len(a_ends) == 2
     b_ends = [e for e in entries if e['activity'] == 'b' and
-                                    e['category'] == 'activity-end']
+              e['category'] == 'activity-end']
     assert len(b_ends) == 1
 
 
+def test_version_act_completed(gitrepo):
+    # This test runs an activity a, undoes a, then runs an activity b that depends on a
+    # During the last run, a should not be rerun since it was already
+    # run the first time.
+    env = builtins.__xonsh_env__
+    dag = env['DAG']
+    a = dag['a'] = Activity(name='a')
+    b = dag['b'] = Activity(name='b', deps={'a'})
+    # run the first time
+    env_main(args=['--activities', 'a', 'x.y.z'])
+    done = compute_activities_completed()
+    assert done == {'a'}
+    # Now that we changed the version the activity is not done
+    env['VERSION'] = 'x.y.zz'
+    done = compute_activities_completed()
+    assert done == set()
+
+    env_main(args=['--activities', 'a', 'x.y.zz'])
```

Comment:
This is still an integration test, and while I am happy to have it, I am looking for something that tests `compute_completed_activities()` directly

Reply:
That is a unit test would not call env_main